### ISSUE TITLE
Seems no support for CF1 OF1 in MultilabelF1

### ISSUE THREAD

**Issue by github-0-searcher** [Lightning-AI/torchmetrics#1665]
Seems there is no support for CF1 and OF1 in MultilabelF1Score.
I use
```python
pre = torch.tensor([
    [1,1,1],
    [0,0,1],
    [1,0,0],
    [0,1,1],
    [0,1,0]
])

gt = torch.tensor([
    [1,0,1],
    [1,1,0],
    [1,1,0],
    [1,0,1],
    [0,1,1]
])
``` 
as an example.

```python
torchmetrics.classification.MultilabelF1Score(
        num_labels=3, 
        average='macro', 
        )(pre, gt)
```
gives 0.5556 and 'micro' also gives 0.5556.

AFAI understand CF1=0.5714 & OF1=0.5556

$$
\begin{aligned}
\mathrm{OP} & =\frac{\sum_i N_i^c}{\sum_i N_i^p}, & \mathrm{CP} & =\frac{1}{C} \sum_i \frac{N_i^c}{N_i^p} \\
\mathrm{OR} & =\frac{\sum_i N_i^c}{\sum_i N_i^g}, & \mathrm{CR} & =\frac{1}{C} \sum_i \frac{N_i^c}{N_i^g} \\
\mathrm{OF} 1 & =\frac{2 \times \mathrm{OP} \times \mathrm{OR}}{\mathrm{OP}+\mathrm{OR}}, & \mathrm{CF} 1 & =\frac{2 \times \mathrm{CP} \times \mathrm{CR}}{\mathrm{CP}+\mathrm{CR}}
\end{aligned}
$$

$C$: number of labels
$N_i^c$: number the number of images that are correctly predicted for the i-th label
$N_i^p$: number of predicted images for the i-th label
$N_i^g$: number of ground truth images for the i-th label.

**Comment by SkafteNicki** [comment]
Just from the formulas you have provided, it is clear that the implementation we have calculates what you call OF1 and what you are missing is CF1. Here a custom implementation of CF1 that just subclasses `MultilabelF1Score`:

```python
class CF1(torchmetrics.classification.MultilabelF1Score):
    def compute(self):
        tp, fp, tn, fn = self._final_state()
        precision = (tp / (tp + fp)).mean()
        recall = (tp / (tp + fn)).mean()
        return 2 * precision * recall / (precision + recall)

cf1 = CF1(num_labels=3)
print(cf1(pre, gt))  # tensor(0.5714)

of1 = torchmetrics.classification.MultilabelF1Score(num_labels=3)
print(of1(pre, gt))  # tensor(0.5556)
```
Hope this helps :)

**Comment by github-0-searcher** [comment]
Sorry for my late reply. 

CF1 and OF1 are metrics for multi-label classification.

CF1 is mean of F1-scores on all classes. And OF1 calculates in a more "marco" way.
I refer you to Sec4.1 in 

> T. Chen, M. Xu, X. Hui, H. Wu and L. Lin, "Learning Semantic-Specific Graph Representation for Multi-Label Image Recognition," 2019 IEEE/CVF International Conference on Computer Vision (ICCV), Seoul, Korea (South), 2019, pp. 522-531, doi: 10.1109/ICCV.2019.00061.

I was looking for a more all-in-one interface for it, and I made a false assumption that 'marco' and 'micro' are OF1 and CF1 respectively.

**Comment by github-0-searcher** [comment]
Its quite a commonly used metric though. And also for OP OR, CP, CR.
